proto/rpc/v1/submit.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package rpc.v1;
+
+// A request to submit the signed transaction to the ledger.
+message SubmitTransactionRequest {
+    // The signed transaction to submit.
+    bytes signed_transaction = 1;
+
+    bool fail_hard = 2;
+}
+
+// A response when a signed transaction is submitted to the ledger.
+message SubmitTransactionResponse {
```

Comment:
In general, the JSON API doesn't have a definition of `Transaction`. Sometimes a transaction is just the bare fields, but it can also have a signature attached, and a hash is attached after it is submitted. 

There's two ways we could go here:
1) just add those fields onto the `Transaction` message and don't populate them until we're ready
2) Pull those fields out of the `Transaction` message and push them to the encompassing message. `SignedTransaction` and `hash` in this message show how this pattern could work. 

In general, I think (1) is the most expedient and keeps consistency with the JSON API. I think (2) is more correct and more in line with what I would expect as a client developer.  Per always, a hard version change in the JSON API could bring these back in line if we go with (2).

Reply:
So I discussed this with the team some time ago, and we decided to just return the hash, and not the transaction itself. Returning the transaction as json and/or a blob is really just for the sign-and-submit mode of this API, which will not be supported via gRPC. Therefore, I see no reason to return the transaction, in any format. The hash can be used to get the status of the transaction via the tx rpc.

Reply:
Right. I apologize for forgetting this conversation. SGTM.